README.md

```diff
@@ -26,39 +27,55 @@ Commands:
 
 awssecrets supports the AWS_* environment variables, and configuration in `~/.aws/`credentials` and `~/.aws/config`
 
-## Example
+## Examples
 
-1. Create a secret in AWS using the CLI
+First create a secret in secrets manager using the AWS CLI tools
 
 ```bash
-aws secretsmanager create-secret --name secret-friendly-name --secret-string secret-string-value
+aws secretsmanager create-secret --name postgresqlRootPwd --secret-string password123DontHackMe
 ```
 
-2. Execute `env` command, all secrets are loaded as environment variables.
+Now execute `env` command, all secrets are loaded as environment variables.
 
 ```bash
-awssecrets -r us-east-1 exec -n secret-name-1 -n secret-name-2 -- env
+awssecrets -r us-east-1 exec -n postgresqlRootPwd -n mysqlRootPwd -- env
+```
+
+\- or -
+
+```Dockerfile
+# how to use within a docker container
+RUN curl -sL \
+  https://github.com/EconomistDigitalSolutions/fcreds/releases/download/v1.0/awssecrets_linux_amd64.tar.gz \
+ | tar zx -C /usr/local/bin \
+ && chmod +x /usr/local/bin/fcreds
+
+# our worker code is `node worker.js` simply prefix
+# exposing it this way as opposed to using an ENV statement means that only the application
+# has the secret not the whole contianer, like a local var as opposed to global scope
+CMD /usr/local/bin/fcreds -r eu-west-2 exec -n postgresqlRootPwd -- node worker.js
 ```
 
 ## IAM policy
 
-To be able to access the secret value, any role used to call fcreds will require an IAM policy that looks like this
-```yaml
+To access the secret value from your application and keep things proper tight use [granular IAM policies](https://docs.aws.amazon.com/secretsmanager/latest/userguide/auth-and-access_identity-based-policies.html#permissions_grant-limited-resources) tied to roles. Any role used to call fcreds will require an IAM policy that looks a bit like this:
+
+```YAML
 - PolicyName: SecretsManagerAccess
   PolicyDocument:
     Statement:
     - Effect: Allow
       Action:
       - secretsmanager:GetSecretValue
       Resource:
-      - arn:aws:secretsmanager:region:accountId:secret:secretsName-*
+      - arn:aws:secretsmanager:region:<accountId>:secret:postgresqlRootPwd
```

Comment:
We actually need the `-*` at the end as the arn will look something like `arn:aws:secretsmanager:<region>:<accountId>:secret:postgresqlRootPwd-ZhqeBy` or we need to add the suffix that AWS adds to the name

Reply:
so could I do `postgresqlRootPwd-*`